plugins/callback/opentelemetry.py

```diff
@@ -542,27 +542,42 @@ def v2_runner_on_failed(self, result, ignore_errors=False):
             status = 'failed'
             self.errors += 1
 
+        # avoid storing task results if it's logs are disabled
+        dump = ""
+        if not self.disable_logs:
+            dump = self._dump_results(result._result)
```

Comment:
Out of curiosity: why not modify `_dump_results` to simply return `''` in case `not self.disable_logs`? Then you only have to modify one place instead of multiple ones.

Reply:
Aha! Thanks @felixfontein, I think that's much simpler.  I'll apply your suggestion

Reply:
Wait, `self._dump_results` is not a function in the callback plugin but something provided by the callback class, https://github.com/ansible/ansible/blob/ac6200b5977a91e0292b0b342db1d4844671d145/lib/ansible/plugins/callback/__init__.py#L200

I don't know if i can override it, but I'll probably create a new function to simplify the logic

Reply:
Ah, in that case, maybe keep your current solution. Or introduce a new function. Both are fine for me :)

(Overwriting that function is probably not a good idea...)

Reply:
see https://github.com/ansible-collections/community.general/pull/8373/commits/e3cd0158d5965e735f47c674f89a0c9790633c2e